homeassistant/components/stream/__init__.py

```diff
@@ -192,19 +205,29 @@ class Stream:
     """Represents a single stream."""
 
     def __init__(
-        self, hass: HomeAssistant, source: str, options: dict[str, str]
+        self,
+        hass: HomeAssistant,
+        source: str,
+        options: dict[str, str],
+        stream_label: str | None = None,
     ) -> None:
         """Initialize a stream."""
         self.hass = hass
         self.source = source
         self.options = options
+        self._stream_label = stream_label
         self.keepalive = False
         self.access_token: str | None = None
         self._thread: threading.Thread | None = None
         self._thread_quit = threading.Event()
         self._outputs: dict[str, StreamOutput] = {}
         self._fast_restart_once = False
         self._available = True
+        self._logger: logging.Logger | logging.LoggerAdapter = _LOGGER
+        if self._stream_label:
+            self._logger = StreamLoggerAdapter(
+                self._logger, {"stream_label": self._stream_label}
+            )
```

Comment:
Why not create a unique logger per stream?

```python
self._logger = logging.getLogger(f"__package__.stream.{stream_label}")
```

Reply:
Done